docs/concepts/source-ip/source-ip.md

```diff
@@ -0,0 +1,251 @@
+client_address=10.240.0.3
+```
+
+Note that these are not your IPs, they're cluster internal IPs. This is what happens:
+
+* Client sends packet to `node2:nodePort`, which doesn't have any endpoints
+* `node2` replaces the source IP on the packet with its own (SNAT)
+* `node2` replaces the destination IP on the packet with the pod IP
+* the pod IP is routed to node 1, and then to the endpoint
+* the pod's reply is routed back to node2
+* the pod's reply is sent back to the client
+
+Visually:
+
+```
+          client
+             \ ^
+              \ \
+               v \
+   node 1 <--- node 2
+    | ^   SNAT
+    | |   --->
+    v |
+ endpoint
+```
+
+
+To avoid this, we have a beta feature in 1.5, triggered by the `service.beta.kubernetes.io/external-traffic`
+[annotation](/docs/user-guide/load-balancer/#loss-of-client-source-ip-for-external-traffic). Setting it to the value `OnlyLocal` will only proxy requests to
+local endpoints, preserving the source IP. If there are no local endpoints,
+packets sent to the node are dropped.
+
+Lets set it and rerun the same test:
+
+```console
+$ kubectl annotate service nodeport service.beta.kubernetes.io/external-traffic=OnlyLocal
+service "nodeport" annotated
+
+$ for node in $NODES; do curl --connect-timeout 1 -s $node:$NODEPORT | grep -i client_address; do
+client_address=104.132.1.79
+```
+
+Note that you only got one reply, with the *right* client IP, from the one node on which the endpoint pod
+is running on.
+
+This is what happens:
+
+* client sends packet to `node2:nodePort`, which doesn't have any endpoints
+* packets are dropped
+* client sends packet to `node1:nodePort`, which *does* have endpoints
+* node1 routes packets to endpoint with the correct source IP
+
+Visually:
+
+```
+        client
+       ^ /   \
+      / /     \
+     / v       X
+   node 1     node 2
+    ^ |
+    | |
+    | v
+ endpoint
+```
+
+
+
+### Source IP for Services with Type=LoadBalancer
+
+As of Kubernetes 1.5, packets sent to Services with [Type=LoadBalancer](/docs/user-guide/services/#type-loadbalancer) are
+source NAT'd by default, because all schedulable Kubernetes nodes in the
+`Ready` state are eligible for loadbalanced traffic. So if packets arrive
+at a node without an endpoint, we need to proxy it to a node *with* an
+endpoint, replacing the source IP on the packet with the IP of the node (as
+described in the previous section).
+
+you can test this by exposing the source-ip-app through a loadbalancer
+
+```console
+$ kubectl expose deployment source-ip-app --name=loadbalancer --port=80 --target-port=8080 --type=LoadBalancer
+service "loadbalancer" exposed
+
+$ kubectl get svc loadbalancer
+NAME           CLUSTER-IP    EXTERNAL-IP       PORT(S)   AGE
+loadbalancer   10.0.65.118   104.198.149.140   80/TCP    5m
+
+$ curl 104.198.149.140
+CLIENT VALUES:
+client_address=10.240.0.5
+...
+```
+
+However, if you're running on GKE/GCE, setting the same `service.beta.kubernetes.io/external-traffic`
```

Comment:
Worth explaining why GCE only - because our LB works as a VIP, rather than a proxy.  We should also explain why the nodeport HCs don't do the same?